modules/renter/contractor/uptime.go

```diff
@@ -0,0 +1,51 @@
+package contractor
+
+import (
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+)
+
+// uptimeInterval specifies how frequently hosts are checked.
+var uptimeInterval = func() time.Duration {
+	switch build.Release {
+	case "dev":
+		return 1 * time.Minute
+	case "standard":
+		return 30 * time.Minute
+	case "testing":
+		return 100 * time.Millisecond
+	}
+	panic("undefined uptimeInterval")
+}()
+
+// threadedMonitorUptime regularly checks host uptime, and deletes contracts
+// whose hosts fall below a minimum uptime threshold.
+func (c *Contractor) threadedMonitorUptime() {
+	for range time.Tick(uptimeInterval) {
+		// get current contracts
+		contracts := c.Contracts()
+
+		// identify hosts with poor uptime
+		badContracts := contracts[:0] // reuse memory during filter
```

Comment:
this is on the boarder of too clever. I see that there are no issues but I had to look more closely to be certain. I think falls outside of the 'obviously correct' standard that we typically aim for.